### ISSUE TITLE
Local Savegames feature not working after updating to 2.1.2

### ISSUE THREAD

**Issue by cvsickle** [Modorganizer2/modorganizer#309]
# The problem

I updated to 2.1.2 from 2.1.1 by unpacking the standalone version into my existing MO2 folder and overwriting everything that needed to be overwritten. MO2 still displays my saves for each profile in the saves tab and those files are in the saves folder in each profile. Skryim won't see those saves and creating a new save results in files being written to the SSE saves folder in My Documents.

# Environment

* Mod Organizer Version that exhibits the issue: 2.1.2
* Last Mod Organizer Version that did not exhibit the issue (if applicable): 2.1.1
* Desktop OS/version used to run Mod Organizer: Windows 10 Pro v1709

# Details

I pretty much always have MO run as administrator. I have tried toggling the Local Savegames checkbox off and back on for each profile.

# Link to Mod Organizer logs

## USVFS
https://gist.github.com/cvsickle/a86aebbbdc7524b6ec490a2368831fd2

## MO Interface
https://gist.github.com/cvsickle/6f0aa93507d25f061b14414c3e9463eb

**Comment by Silarn** [comment]
Are you sure anything is working? Your VFS does not appear to be hooking properly. Please make sure your files are unblocked (usually unblocking the zip first unblocks all of the files) and excluded from any AV and that all of your relevant skse plugins are updated. Can you replicate this running the game directly or another tool that parses saves files?

**Comment by cvsickle** [comment]
All my mods seem to be loading correctly, both in the game and in xEdit. I tried unblocking the zip and re-extracting, but no change. I just updated all my skse plugins when 2.00.07 was released the other day and everything seemed fine then. I'm not sure what you mean by running the game directly. Do you mean not through skse, or not through MO2 in general?

**Comment by cvsickle** [comment]
I created an empty profile in MO2 and only installed Alternate Start. I ran SSE on its own without SKSE and started a new game. The save from that game ended up in the My Documents folder.

**Comment by cvsickle** [comment]
Yeah saves are not loading or being created in the profiles, they're only showing up in/from the global folder in My Documents.

If something else isn't working, I haven't noticed it yet. Could it be the way I updated it?

**Comment by Silarn** [comment]
Alright, that's kind of odd. The save mapping should work just the same as before. Can you try moving your saves out of your profile temporarily, backup your profile and my documents settings files, let the game regenerate those files. Reapply the local saves (rename and move back into the profile) and then see what happens?

Running a tool like Wrye Bash should also try to load your saves and might be a faster way to verify they're being rerouted properly.

Basically my main guess is the game settings are not being updated to look into the _Mo Saves directory in the settings file. The VFS is supposed to reroute this directory, but if the game is still looking in the standard location then it won't see you virtual files.

**Comment by Silarn** [comment]
The other alternative is something had broken local settings and the save directory change is only getting applied to the documents settings file.

**Comment by cvsickle** [comment]
Ok, I'm not 100% sure if I did exactly what you are saying, but here's what I did.

I backed up all my files. I deleted the skyrim.ini and skyrimprefs.ini from my documents. I ran skyrim via steam (not MO) to let it create those two files. I copied those files into my empty profile folder in MO2 (mentioned above). In MO2, I unchecked and rechecked Local Savegames and Local Game Settings for the empty profile. I ran SSE from MO2 (not skse) and created a new save. That save showed up in the my documents folder and is not visible in the saves tab in MO2.

MO Interface Log: https://gist.github.com/cvsickle/6ea5c3671138d6bb99ae3bbc83275dab
USVFS Log: https://gist.github.com/cvsickle/5a2024e9d872840a402a04839b5ec835

Is this what you were suggesting?

**Comment by cvsickle** [comment]
I've also tried to revert to 2.1.1 in a similar manner to how I upgraded. 2.1.1 is doing the same thing with the global saves, so I'm pretty sure something broke when I was upgrading.

**Comment by Silarn** [comment]
The problem is the correct setting is not being placed in your Skyrim.ini file. It needs to redirect the save location to __MO_Saves. Delete any customsave.ini in your profile and add that custom save location to the local ini. The local saves will now work.

This should be done automatically, and I'm not sure why it isn't.

**Comment by Silarn** [comment]
Specifically this is
```ini
[General]
SLocalSavePath=__MO_Saves\
```

**Comment by Silarn** [comment]
I guess you should try that. But Skyrim.ini being in your profile is all that's needed to enable local settings (and trigger the mapping of these files) - and I see in your log that inibakery is being loaded (which handles the ini file mappings). If that setting is present in your local Skyrim.ini and the saves directory is also present (without a _) then it should be working.

**Comment by cvsickle** [comment]
So I've proven to myself that the profile ini files are not being used. Both my global and profile skyrim.ini had bDisableGearedUp=1 in them, which prevents your favorite gear from showing up as equipped on your 3rd person character model.

I set that to 0 in my profile ini, and my extra favorite gear would still not show up. I set it back to 1 in the profile ini and then set it to 0 in my global ini and the extra favorite gear showed up.

So those profile ini files are not being used, hence why the SLocalSavePath value is not being used either.

**Comment by octatone** [comment]
Seeing the same behavior with local save games not working.  Saves in new profiles, are going to the default profile which is not set to use local save games, which means they are going into Documents/My Games/Game

**Comment by cvsickle** [comment]
I guess I haven't mentioned yet that this has happened to me before on an older version. I think I was on the one before 2.1, but it's been awhile. I just thought it was weird that it was working fine until I updated.

**Comment by cvsickle** [comment]
Here's a question that might be related somehow. Why is MO acting like it's an instanced application when I wanted it to be local? I installed it by extracting the standalone archive, but why is it writing data to its AppData folder? All of the logs are being written to AppData and not the logs folder in my MO2 install directory. ModOrganizer.ini is only found in AppData. Adding a splash.png only works when adding it to AppData - not from the install folder. I'm thinking I should reinstall using the installer version and make it a local only.

**Comment by Al12rs** [comment]
You probably mean "portable" not local. Local is generally used to mean the local pc, in fact programs store their files in the Appdata/local folder. What you want is a portable "instance" not a portable install. The standalone download just means that you don't have to use an installer and download additional files to get the Mo2 installation, the standalone is just the preinstalled Mo2 folder. You can manage and change instances with the top left button on the Mo2 main windows. There you can choose postable intance to make it so Mo completely avoids using Appdata, with the disadvantage that you can only have one portable instance and any other installation of Mo2 will not be able to see that one. You could instead keep the "Global" instance and simply move the path of the base directory away from appdata in the Mo2 settings. That way only the bare minimum will still go in Appdata, nominally the Mo2 INI (to know where you set your paths), the crash dumps and logs since those need to be written even before Mo actually reads the INI file.

**Comment by octatone** [comment]
Why was this closed.  Is it actually fixed?

**Comment by Al12rs** [comment]
It's not?

**Comment by octatone** [comment]
The discussion here digressed into the topic of portable mode which wasn't related to the initial bug report. I'll check again with the latest build if local saves are working.

**Comment by cvsickle** [comment]
Can confirm everything is working for me on the current version (portable and local game settings and local save games).

**Comment by Silarn** [comment]
For the VR games this was largely due to the core ini not getting added to the game plugin to be managed/virtualized. So local settings would break it, and it was writing the necessary changes to the wrong INI as it seemed that the save directory change was only working in the 'main' ini file for some reason.

Other games shouldn't have that issue, but can occasionally get stuck in a bad state. Best thing to do then is either manually fix the settings or let them regenerate from the defaults.

**Comment by Normal69** [comment]
> 
> 
> For the VR games this was largely due to the core ini not getting added to the game plugin to be managed/virtualized. So local settings would break it, and it was writing the necessary changes to the wrong INI as it seemed that the save directory change was only working in the 'main' ini file for some reason.
> 
> Other games shouldn't have that issue, but can occasionally get stuck in a bad state. Best thing to do then is either manually fix the settings or let them regenerate from the defaults.

Hmm, so where could I fix this one, please?
I have some ancient installs I won't like to ruin, Skyrim SE portable, Skyrim 32 and Fallout4 ones.

Currently my Skyrim SE using the settings and saves in c:\Users\Normal\Documents\My Games\Skyrim Special Edition\ which I'd like to avoid.

Tried all of the combinations written above with no success.
Where is that elusive ini I need to modify, or at least what is the name to search for?

I'm on MO2 2.4.2.

Just your Normal thread necromancer

**Comment by Normal69** [comment]
Okay, went over the discord, and somehow running the game once with MO2 debug activated fixed it.
"Go in the Settings, Diagnostics tab, set Log Level to Debug."

**Comment by stier8005-sys** [comment]
try clearing shader cache.
